packages/react/src/primitives/shared/datastore.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ModelPredicate,
+  PersistentModel,
+  ProducerModelPredicate,
+} from '@aws-amplify/datastore';
+import { isFunction } from './utils';
+
+export type DataStorePredicateObject = {
+  and?: DataStorePredicateObject[];
+  or?: DataStorePredicateObject[];
+  field?: string;
+  operand?: string;
+  operator?: string;
+};
+
+/**
+ * Given an array of predicates, compose them in sequential order
+ */
+const mergePredicates = <Model extends PersistentModel>(
+  predicates: ProducerModelPredicate<Model>[]
+): ProducerModelPredicate<Model> =>
+  predicates.reduce(
+    (previous, current) => (predicate) => current(previous(predicate)),
+    (predicate) => predicate
+  );
```

Comment:
This function is a bit confusing to me. Does it return a function that accepts a value and returns a boolean indicating whether that value passes all composed predicates?

Reply:
from a list of predicates, will return a composed version:

`f(x), g(x), h(x) => f(g(h(x)))`

Reply:
I see, I guess I just haven't seen predicates composed like this. Ultimately, a predicate will convert whatever type of value it's given into a `boolean` so, in your example above, `f` and `g` would be accepting a `boolean` as their argument instead of the original type of the parameter `x`.